chall04/pcuadrad.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import sys
+
+def error_message(filename, message):
+    print (sys.argv[0] + ": " + filename + ": " + message)
+
+def place_books(books, shelfs):
+    rest = 0
+    last_book = 0
+    for i in range(0, len(shelfs)):
+        begin = shelfs[i]
+        begin += rest
+        rest = 0
+        if begin < 0 and i is len(shelfs) - 1:
+            return False
+        for j in range(last_book, len(books)):
+            begin -= books[j]
+            if begin < 0:
+                rest = begin
+                last_book = j + 1
+                break
+            if j is len(books) - 1:
+                print (i + 1)
+                return True
+    print (i + 1)
+    return True
+
+def parse_file(txt):
+    i = 0
+    books = []
+    shelfs_int = []
+    for line in txt:
+        if i is 0:
+            shelfs = line.split(' ')
+        else:
+            line2 = line.split(' ', 1)
+            books.append(int(line2[0]))
+        i += 1
+    [shelfs_int.append(int(i)) for i in shelfs]
+    shelfs_int = sorted(shelfs_int, reverse=True)
+    books = sorted(books, reverse=True)
```

Comment:
If you already have the `books` list, doing assigning `sorted` will copy the list then sort it.  
It would have been more optimized to simply call the `sort` method on the list:
```suggestion
    books.sort(reverse=True)
```
You don't have to assign it, the elements will be moved inside the list, and no copy will be done.